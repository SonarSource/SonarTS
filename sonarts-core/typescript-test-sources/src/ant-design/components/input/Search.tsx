import React from 'react';
import classNames from 'classnames';
import Input, { HTMLInputProps } from './Input';
import Icon from '../icon';

export interface SearchProps {
  className?: string;
  placeholder?: string;
  prefixCls?: string;
  style?: React.CSSProperties;
  defaultValue?: any;
  value?: any;
  onSearch?: (value: string) => any;
  size?: 'large' | 'default' | 'small';
  disabled?: boolean;
  readOnly?: boolean;
}

export default class Search extends React.Component<SearchProps & HTMLInputProps, any> {
  static defaultProps = {
    prefixCls: 'ant-input-search',
    onSearch() {},
  };
  input: any;
  onSearch = () => {
    const { onSearch } = this.props;
    if (onSearch) {
      onSearch(this.input.refs.input.value);
    }
    this.input.refs.input.focus();
  }
  render() {
    const { className, prefixCls, ...others } = this.props;
    delete (others as any).onSearch;
    const searchSuffix = (
      <Icon
        className={`${prefixCls}-icon`}
        onClick={this.onSearch}
        type="search"
      />
    );
    return (
      <Input
        onPressEnter={this.onSearch}
        {...others}
        suffix={searchSuffix}
        className={classNames(prefixCls, className)}
        ref={node => this.input = node}
      />
    );
  }
}
