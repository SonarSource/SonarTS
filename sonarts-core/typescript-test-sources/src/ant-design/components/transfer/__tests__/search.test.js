import React from 'react';
import { mount } from 'enzyme';
import Search from '../search';

describe('Search', () => {
  it('should show cross icon when input value exists', () => {
    const wrapper = mount(<Search value={''} />);

    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ value: 'a' });

    expect(wrapper).toMatchSnapshot();
  });
});
