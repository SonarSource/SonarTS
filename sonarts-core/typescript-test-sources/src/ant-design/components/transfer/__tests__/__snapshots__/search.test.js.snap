// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should show cross icon when input value exists 1`] = `
<Search
  placeholder=""
  value=""
>
  <div>
    <Input
      autosize={false}
      disabled={false}
      onChange={[Function]}
      placeholder=""
      prefixCls="ant-input"
      type="text"
      value=""
    >
      <input
        className="ant-input"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        type="text"
        value=""
      />
    </Input>
    <span
      className="undefined-action"
    >
      <Icon
        type="search"
      >
        <i
          className="anticon anticon-search"
        />
      </Icon>
    </span>
  </div>
</Search>
`;

exports[`Search should show cross icon when input value exists 2`] = `
<Search
  placeholder=""
  value="a"
>
  <div>
    <Input
      autosize={false}
      disabled={false}
      onChange={[Function]}
      placeholder=""
      prefixCls="ant-input"
      type="text"
      value="a"
    >
      <input
        className="ant-input"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        type="text"
        value="a"
      />
    </Input>
    <a
      className="undefined-action"
      href="#"
      onClick={[Function]}
    >
      <Icon
        type="cross-circle"
      >
        <i
          className="anticon anticon-cross-circle"
        />
      </Icon>
    </a>
  </div>
</Search>
`;
